interfaces/builtin/x11.go

```diff
@@ -153,6 +156,56 @@ type x11Interface struct {
 	commonInterface
 }
 
+func (iface *x11Interface) MountConnectedPlug(spec *mount.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
+	switch slotSnapName := slot.Snap().InstanceName(); slotSnapName {
+	case "ubuntu-core", "core", "snapd":
```

Comment:
Could you use implicitSystemConnectedSlot(slot) here?  That would avoid adding more ways of detecting implicit slots.

Reply:
Done